### ISSUE TITLE
Replay marshaled race

### ISSUE THREAD

**Issue by HazardCreative** [RotorHazard/RotorHazard#271]
Add the ability to "replay" a race that has been adjusted within RH, so that the times can be sent to services that can't access RH's API or database, such as LiveTime or Rotormatch.

**Comment by HazardCreative** [comment]
I've discovered a complicating factor. LiveTime requests the current time when it triggers the race start. But RH has some setup to do before the server starts the race, which currently takes about 1.5s. We need to find a way to store the time supplied to LiveTime before the race kickoff. This could either be stored in the race data, or by overwriting the actual race start time. I do not know what other systems require for recording start time, so more research is needed. 

Since RH uses and stores monotonic time and LT uses epoch time, supplying times to LT will also be incorrect whenever the server restarts. To survive a restart and be able to supply times that LiveTime understands, the race epoch time would also need recorded. This might be a non-issue since RH's database does not currently store race data when LiveTime is driving—but without doing so LT can only make use of retroactive calibration adjustments until the next race begins.

**Comment by pulquero** [comment]
I have a branch somewhere where I did some work on unifying different timing approaches. Sent from my Galaxy
-------- Original message --------From: Hazard Creative <notifications@github.com> Date: 15/11/2020  03:50  (GMT+00:00) To: RotorHazard/RotorHazard <RotorHazard@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [RotorHazard/RotorHazard] Replay marshaled race (#271) 

I've discovered a complicating factor. LiveTime requests the current time when it triggers the race start. But RH has some setup to do before the server starts the race, which currently takes about 1.5s. We need to find a way to store the time supplied to LiveTime before the race kickoff. This could either be stored in the race data, or by overwriting the actual race start time. I do not know what other systems require for recording start time, so more research is needed.

Since RH uses and stores monotonic time and LT uses epoch time, supplying times to LT will also be incorrect whenever the server restarts. To survive a restart and be able to supply times that LiveTime understands, the race epoch time would also need recorded. This might be a non-issue since RH's database does not currently store race data when LiveTime is driving—but without doing so LT can only make use of retroactive calibration adjustments until the next race begins.
